datafusion/common/src/config.rs

```diff
@@ -270,7 +270,48 @@ config_namespace! {
         /// will be reordered heuristically to minimize the cost of evaluation. If false,
         /// the filters are applied in the same order as written in the query
         pub reorder_filters: bool, default = false
+
+        // The following map to parquet::file::properties::WriterProperties
+
+        /// Sets best effort maximum size of data page in bytes
+        pub data_pagesize_limit: usize, default = 1024 * 1024
```

Comment:
It is an interesting question if we want these settings to be session level (as this change proposes) or if they should be per-sql level 🤔 

I suppose having a session level default would make sense and if we want to add per statement overrides (like `COPY TO <file> AS PARQUET (DATA_PAGE_ROW_COUNT_LIMIT 100000)`) we can always do that afterwards as well.

Reply:
I think we will definitely want `COPY TO` to be able to set any of these configs on a per statement basis.

For `insert into`, we could allow the table itself to be registered with specific settings e.g.:

```sql
create external table my_table(x int, y int) 
stored as parquet
location '/tmp/my_table' 
WITH (
DATA_PAGESIZE_LIMIT 2048,
DATA_PAGE_ROW_COUNT_LIMIT 100000)
...
);
```

`insert into mytable` would then use any table specific settings or fall back to the session level configs.